File: java-concert/src/main/java/prac/javaconcert/concert/domain/Concert.java
package prac.javaconcert.concert.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Entity
public class Concert {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;


    // 콘서트 이름
    private String name;


    // 콘서트 포스터
    private String poster;

    // 콘서트 장소
    @OneToOne(mappedBy = "concert")
    private Place place;

    // 콘서트 기간
    private String period;

    // 콘서트 시간
    private Integer runningTime;

    // 콘서트 관람 연령
    private String ageLimit;

    // 콘서트 장르
    @Enumerated(EnumType.STRING)
    private Genre genre;

    // 콘서트 정보
    private String description;

    public Concert(String name, String poster, String period, Integer runningTime, String ageLimit, Genre genre, String description) {
        this.name = name;
        this.poster = poster;
        this.period = period;
        this.runningTime = runningTime;
        this.ageLimit = ageLimit;
        this.genre = genre;
        this.description = description;
    }
}

File: concert/src/main/kotlin/prac/concert/domain/Concert.kt
package prac.concert.domain

import jakarta.persistence.*


@Entity
class Concert(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id : Long? = null,

    @Column
    val name : String,


    @Column
    val poster : String,


    @OneToOne(mappedBy = "concert", fetch = FetchType.LAZY)
    var place : Place? = null,


    @Column
    val period : String,

    @Column
    val runningTime : Int,

    @Column
    val ageLimit : String,


    @Column
    @Enumerated(EnumType.STRING)
    val genre : Genre,

    @Column
    val description : String,
){
    constructor(
        name: String,
        poster: String,
        period: String,
        runningTime: Int,
        ageLimit: String,
        genre: Genre,
        description: String
    ) : this(
        id = null,
        name = name,
        poster = poster,
        period = period,
        runningTime = runningTime,
        ageLimit = ageLimit,
        genre = genre,
        description = description
    )
}